lib/daru/index.rb

```diff
@@ -330,5 +341,11 @@ def values
     def inspect
       "Daru::MultiIndex:#{self.object_id} (levels: #{levels}\nlabels: #{labels})"
     end
+
+    def factor_out
+      # Remove levels not needed for display
+      useless = levels.index { |l| l.length > 1 }
+      drop_left_level useless
+    end
```

Comment:
You just found a bug!

Yes you will absolutely have to pass the number of levels to be dropped because when a user specifies a given number of indexes for a MultiIndex vector/dataframe, only those many levels should be dropped.

Your previous approach worked mainly because the specs were so written. What if the levels were like this:
``` ruby
levels: [[:a], [:one], [:one, :two, :three, :four]]
```
And I have specified only `v[:a]` when calling the vector?
Don't rely only on the tests, they're there just as a reference.